src/models/fields/base-field-configuration.ts

```diff
@@ -1,38 +1,241 @@
 import Joi from 'joi';
 import { schema as baseFieldSchema, BaseField } from './base-field';
+import {
+  stringOrTranslationsSchema,
+  StringOrTranslations,
+} from '../../translations';
 
 const schema = Joi.object().keys({
   key: Joi.string().pattern(/^[a-z][a-zA-Z\d]*$/).required().example('id'),
-  name: Joi.string().required().example('ID as used in our geographic information system'),
-  inputType: Joi.string().valid('text', 'textarea', 'select', 'radio', 'switch', 'checkbox', 'file', 'files')
+  type: Joi.string().valid('string', 'boolean', 'number', 'integer')
```

Comment:
These `.when` constructs are a bit hard to validate, as only the first one that matches is evaluated. Why did you opt for this instead of for the `Joi.alternatives()` mechanism that matches the TS interface?

Reply:
I've tried! It is not working :( 

It throws error that it cannot find any alternative for such object (!). And shows that schema with applied defaults, taken from one of alternative schema. I've spent I think hours just on trying to make it work, rewriting that alternative schemas  again and again with same result.

Reply:
From terminal tab history

```
Body: "environment.fieldConfigurations.pinGroups[0]" does not match any of the allowed types (HTTP 500)
```
......
```
      "pinGroups": [
        {
          "key": "assetId",
          "hint": "",
          "name": "Meetpaalnummer",
          "valueOptions": null,
          "type": "string",
          "defaultValue": "",
          "inputType": "text"
        },
```

in DB it has only `key`, `name` and `hint` ! All these defaults taken from **alternative** schema, expected one to be applied! 

But - you see result on top of this comment.

Reply:
https://github.com/withthegrid/platform-sdk/blob/28554e66f460662e94a9c00fa94e412375988d11/src/models/fields/base-field-configuration.ts#L27

here is that Joi alternatives schema.

Reply:
If you print the validation error (using JSON.stringify), you'll get an extensive explanation about why it did not match. The relevant part of it:
```
{
  "message": "\"defaultValue\" is not allowed to be empty",
  "path": [
    "defaultValue"
  ],
  "type": "string.empty",
  "context": {
    "label": "defaultValue",
    "value": "",
    "key": "defaultValue"
  }
},
```

The issue is a very bad legacy decision by Joi to not allow an empty string by default. You have to add `.allow('')` after the `defaultValue` schema and then it validates like a charm :).

Note to self, it took me a while to get it running, but this is the solution: `ts-node --project ./tsconfig.cjs.json ./src/test`

Reply:
> you'll get an extensive explanation about why it did not match

Oh, thought that I have already everything possible at console :(

Great then! Will revert back to alternatives now!